homeassistant/components/media_player/frontier_silicon.py

```diff
@@ -0,0 +1,245 @@
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.frontier_silicon/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    SUPPORT_NEXT_TRACK, SUPPORT_PAUSE, SUPPORT_PREVIOUS_TRACK, SUPPORT_SEEK,
+    SUPPORT_PLAY_MEDIA, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    SUPPORT_VOLUME_STEP, SUPPORT_STOP, SUPPORT_TURN_OFF, SUPPORT_TURN_ON,
+    SUPPORT_PLAY, SUPPORT_SELECT_SOURCE, MediaPlayerDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    STATE_OFF, STATE_PLAYING, STATE_PAUSED, STATE_UNKNOWN,
+    CONF_HOST, CONF_PORT, CONF_PASSWORD)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['fsapi==0.0.7']
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_FRONTIER_SILICON = SUPPORT_PAUSE | SUPPORT_VOLUME_SET | \
+    SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP | \
+    SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | SUPPORT_SEEK | \
+    SUPPORT_PLAY_MEDIA | SUPPORT_PLAY | SUPPORT_STOP | SUPPORT_TURN_ON | \
+    SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE
+
+DEFAULT_PORT = 80
+DEFAULT_PASSWORD = '1234'
+DEVICE_URL = 'http://{0}:{1}/device'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_PASSWORD, default=DEFAULT_PASSWORD): cv.string,
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Frontier Silicon platform."""
+    import requests
+
+    if discovery_info is not None:
+        add_devices(
+            [FSAPIDevice(discovery_info, DEFAULT_PASSWORD)],
+            update_before_add=True)
+        return True
+
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    password = config.get(CONF_PASSWORD)
+
+    try:
+        add_devices(
+            [FSAPIDevice(DEVICE_URL.format(host, port), password)],
+            update_before_add=True)
+        _LOGGER.debug('FSAPI device %s:%s -> %s', host, port, password)
+        return True
+    except requests.exceptions.RequestException:
+        _LOGGER.error('Could not add the FSAPI device at %s:%s -> %s',
+                      host, port, password)
+
+    return False
+
+
+class FSAPIDevice(MediaPlayerDevice):
+    """Representation of a Frontier Silicon device on the network."""
+
+    def __init__(self, device_url, password):
+        """Initialize the Frontier Silicon API device."""
+        self._device_url = device_url
+        self._password = password
+        self._state = STATE_UNKNOWN
+
+        self._name = None
+        self._title = None
+        self._mute = None
+        self._source = None
+        self._source_list = None
+        self._media_image_url = None
+
+    def get_fs(self):
+        """Create a fsapi session."""
+        from fsapi import FSAPI
+
+        return FSAPI(self._device_url, self._password)
```

Comment:
I'm OK with that behavior then. Just a few more thoughts:

1) Can you include a short version of this explanation in the function docstring? That way there is an explanation showing that this is being done by design.

2) You should consider making this a property for readability. That way we can document in one place why it is recreated for each request, but we don't have to get bogged down with the details at each usage. The preference in Python is to use properties in favor of get_* method patterns.

```python
@property
def fs_device(self):
    """Create a fsapi session.

    Explanation text here ...
    """
    from fsapi import FSAPI

    return FSAPI(self._device_url, self._password)

# Example:
def media_play(self):
    """Send play command."""
    self.fs_device.play()
```